src/arch.rs

```diff
@@ -255,3 +286,144 @@ pub enum SingleStepGdbBehavior {
+    pub alt_name: Option<&'a str>,
+    /// Size in bits of a register.
+    pub bitsize: usize,
+    /// The offset within the 'g' and 'G' packet of the register data for this
+    /// register.
+    pub offset: usize,
+    /// The encoding type of the register.
+    pub encoding: Encoding,
+    /// The preferred format for display of this register.
+    pub format: Format,
+    /// The register set name this register belongs to.
+    pub set: &'a str,
+    /// The GCC compiler registers number for this register.
+    pub gcc: Option<usize>,
+    /// The DWARF register number for this register that is used for this
+    /// register in the debug information.
+    pub dwarf: Option<usize>,
+    /// Specify as a generic register.
+    pub generic: Option<Generic>,
+    /// Other concrete register values this register is contained in.
+    pub container_regs: Option<&'a [usize]>,
+    /// Specifies which register values should be invalidated when this register
+    /// is modified.
+    pub invalidate_regs: Option<&'a [usize]>,
+}
+
+/// (LLDB extension) Describes the encoding type of the register.
+#[non_exhaustive]
+pub enum Encoding {
+    /// Unsigned integer
+    Uint,
+    /// Signed integer
+    Sint,
+    /// IEEE 754 float
+    IEEE754,
+    /// Vector register
+    Vector,
+}
+
+/// (LLDB extension) Describes the preferred format for display of this
+/// register.
+#[non_exhaustive]
+pub enum Format {
+    /// Binary format
+    Binary,
+    /// Decimal format
+    Decimal,
+    /// Hexadecimal format
+    Hex,
+    /// Floating point format
+    Float,
+    /// 8 bit signed int vector
+    VectorSInt8,
+    /// 8 bit unsigned int vector
+    VectorUInt8,
+    /// 16 bit signed int vector
+    VectorSInt16,
+    /// 16 bit unsigned int vector
+    VectorUInt16,
+    /// 32 bit signed int vector
+    VectorSInt32,
+    /// 32 bit unsigned int vector
+    VectorUInt32,
+    /// 32 bit floating point vector
+    VectorFloat32,
+    /// 128 bit unsigned int vector
+    VectorUInt128,
+}
+
+/// (LLDB extension) Describes the generic types that most CPUs have.
+#[non_exhaustive]
+pub enum Generic {
+    /// Program counter register
+    Pc,
+    /// Stack pointer register
+    Sp,
+    /// Frame pointer register
+    Fp,
+    /// Return address register
+    Ra,
+    /// CPU flags register
+    Flags,
+    /// Function argument 1
+    Arg1,
+    /// Function argument 2
+    Arg2,
+    /// Function argument 3
+    Arg3,
+    /// Function argument 4
+    Arg4,
+    /// Function argument 5
+    Arg5,
+    /// Function argument 6
+    Arg6,
+    /// Function argument 7
+    Arg7,
+    /// Function argument 8
+    Arg8,
+}
+
+/// (LLDB extension) If your target is using the `qRegisterInfo` packet, this
+/// token hast to be returned from the `register_info` function to guarantee
+/// that the callback function to send the register info has been invoked.
+pub struct CallbackToken<'a>(pub(crate) core::marker::PhantomData<&'a *mut ()>);
```

Comment:
`Callback` and `CallbackToken` can move back into the `target` module, as they aren't used by the `Arch`-level implementation of this functionality.